core/planner.c

```diff
@@ -764,23 +767,23 @@ bool plan(struct deco_state *ds, struct diveplan *diveplan, struct dive *dive, i
 	vpmb_start_gradient(ds);
 	if (decoMode() == RECREATIONAL) {
 		bool safety_stop = prefs.safetystop && max_depth >= 10000;
-		track_ascent_gas(depth, &dive->cylinder[current_cylinder], avg_depth, bottom_time, safety_stop, divemode);
+		track_ascent_gas(depth, get_cylinder(dive, current_cylinder), avg_depth, bottom_time, safety_stop, divemode);
```

Comment:
Let's make this take an index rather than a pointer.

Reply:
Or even better, make current cylinder a copied struct cylinder_t rather than a pointer.

Reply:
I don't get it. `track_ascent_gas()` does not store a reference to the cylinder, does it? Moreover it seems to modify the cylinder. So you *have* to pass in a pointer to the cylinder. Using a copy would defeat the purpose or am I missing something?

What I can imagine has to be done is add checks that current_cylinder is still valid after the user deleted cylinders.